s/elements/bindings-editor/utils/prepare-assign-keybind.ts

```diff
@@ -0,0 +1,27 @@
+
+import {AssignKeybind} from "../types.js"
+import {Bindings} from "../../../types.js"
+import {NubContext} from "../../context/element.js"
+
+export function prepareAssignKeybind(context: NubContext): AssignKeybind {
+	return function assignKeyBind(
+			action: string,
+			keyIndex: number,
+			code: string,
+		) {
+
+		const bindings = <Bindings>structuredClone(context.getBindings())
```

Comment:
look at this `structuredClone` function -- i think this is basically an efficient deep-copy browser builtin function